### ISSUE TITLE
If no ephemeris output from file or from A+R exit out gracefully

### ISSUE THREAD

**Issue by mschwamb** [dirac-institute/sorcha#619]
If no ephemeris output from file or from A+R then raise error and exit out - currently the code does not die gracefully

**Comment by mschwamb** [comment]
Maybe it's not an error because it could be you inputted all things in the northern hemisphere, but it would be good that the code gracefully exit as there are no observations to process

**Comment by astronomerritt** [comment]
We don't want to exit the code completely - it's possible that one chunk has no ephemeris output, but the next chunk might. So Sorcha should log a warning that the ephemeris is empty for this chunk, then skip to the next chunk.

**Comment by drewoldag** [comment]
It seems like what is required for this issue is actually already implemented here: https://github.com/dirac-institute/sorcha/blob/main/src/sorcha/sorcha.py#L192-L199

We just need to move the call to `PPMatchPointingToObservations` to a point after the length check of the observations dataframe.

**Comment by mschwamb** [comment]
I think so. I wasn't sure if there was ever a way the merge could fail in PPMatchPointingToObservations where you would also need to check so I though those lines might need to be duplicated and put after ephemeris generation/readin

**Comment by drewoldag** [comment]
I'm pretty sure this is safe. Since the `observations` dataframe always contains a FrameId from the `pointings` dataframe, there should always be _something_ to merge on and therefore _something_ that comes out of `PPMatchPointingToObservations`. 

But I'll think some more about this tomorrow.

**Comment by astronomerritt** [comment]
I also think this is okay. I can't think of a scenario where `PPMatchPointingToObservations` would result in an empty dataframe if there's ephemeris output for that chunk. Even in the edge case where the user is using their own ephemeris file, and generated it using a different pointing database than the one they've provided to Sorcha (so none of the entries match), `PPMatchPointingToObservations` will error out gracefully as the observation times won't match.

I did think I'd already implemented error-handling for this situation! If I recall correctly, Grigori's mini-moons quite often would have no ephemeris results for a whole chunk so I popped that condition in for him. It definitely used to work, but things get shuffled about. :)
